filament_scale_enhanced/__init__.py

```diff
@@ -70,18 +98,210 @@ def get_update_information(self):
+        if self.scale:
+            scale_type = self._settings.get(["scale_type"])
+            self._logger.debug(f"Scale deinit: [{scale_type}]")
+            self.scale.deinit()
+            self.scale = None
+
+    def scale_check_timer_init(self) -> None:
+        """Inits scale check RepeatedTimer if scale exists"""
+        if self.scale:
+            timer_delay = float(self._settings.get(["update_delay"]))
+            self._logger.debug(f"ui_refresh_timer init, delay: [{timer_delay}]")
+            self.ui_refresh_timer = RepeatedTimer(timer_delay, self.check_weight)
+            self.ui_refresh_timer.start()
+        else:
+            self.scale_check_timer_deinit()
+            self._logger.error("RepatedTimer [check_weight()]: no scale present")
+
+    def scale_check_timer_deinit(self) -> None:
+        """Deinits scale check RepeatedTimer if ui_refresh_timer exists"""
+        if self.ui_refresh_timer:
+            self._logger.debug("ui_refresh_timer deinit")
+            self.ui_refresh_timer.cancel()
+            self.ui_refresh_timer = None
+
+    # API Helpers
+
+    def scale_tare(self) -> float:
+        current_offset = self._settings.get(["offset"])
+        self._logger.info(f"Tare started, current offset: [{current_offset}]")
+
+        new_offset = self.scale.tare()
+
+        self._settings.set(["offset"], new_offset)
+        self._settings.save()
+        self._logger.info(f"Tare complete, new_offset: [{new_offset}]")
+
+        return new_offset
+
+    def scale_calibrate(self, weight: float) -> float:
+        current_cal_factor = self._settings.get(["cal_factor"])
+        self._logger.info(f"Cal start, weight: [{weight}], current_cal_factor: [{current_cal_factor}]")
+        new_cal_factor = self.scale.calibrate(weight)
+
+        self._settings.set(["cal_factor"], new_cal_factor)
+        self._settings.save()
+        self._logger.info(f"Cal complete, weight: [{weight}], new_cal_factor: [{new_cal_factor}]")
+
+        return new_cal_factor
+
+    def scale_weight(self) -> float:
+        self._logger.debug(f"Weight last_value: [{self.scale.last_value}], last_value_time: [{self.scale.last_value_time}]")
+
+        update_delay = self._settings.get_float(["update_delay"])
+        if time.time() - self.scale.last_value_time < (update_delay - 0.1):
+            weight = self.scale.last_value
+        else:
+            weight = self.scale.get_weight()
+
+        self._logger.info(f"Scale weight: [{weight}]")
+
+        return weight
+
+    # REST API Handling
+
+    # pylint: disable=too-many-return-statements,too-many-branches,too-many-statements,broad-except
+    def on_api_get(self, request) -> flask.Response:
+        if time.time() - self.last_api_time < 0.1:
+            return flask.make_response("Slow down!", 200)
+
+        command = request.args.get("command")
+        value = request.args.get("value")
+
+        if command == "tare":
+            self.last_api_time = time.time()
```

Comment:
A method I often use here is
```python
getattr(self, command)()
```

This makes adding new functionality just adding a method, keeps the entry function smaller, and makes unittesting easier.

As it all involves hardware, unit testing is probably hard. But we might be able to write a mock scale class. But I wouldn't consider that level required to merge.